sdk/core/azure-core/src/main/java/com/azure/core/http/HttpPipeline.java

```diff
@@ -103,7 +104,7 @@ public Mono<HttpResponse> send(HttpPipelineCallContext context) {
         return Mono.defer(() -> {
             HttpPipelineNextPolicy next =
                 new HttpPipelineNextPolicy(new HttpPipelineCallState(this, context));
-            return next.process();
+            return next.process().publishOn(Schedulers.boundedElastic());
```

Comment:
Should we consider moving this `publishOn` to `this.pipeline.getHttpClient().send(this.context.getHttpRequest(), this.context.getContext())` in `HttpPipelineNextPolicy` since we want to do this at the HTTP client level for the network operation?

Reply:
This is a good question on where the `publishOn` should be applied. This could be pushed further down the call stack to where the request is being sent over the wire, which is what you called out in it being applied to `send`, or this could be pulled further up the call stack to when the processing first leaves the specific SDK into Core, being placed somewhere in `RestProxy`. The latter offloads all serialization and deserialization to the elastic thread.

Reply:
There's a possibility that there's a custom HTTP pipeline policy that might potentially override the `publishOn` if we have it at this level. So, we could do that in RestProxy (or in HttpPipeline) but we should still have `publishOn` at the client level too where the request is sent over the wire just to ensure no other policy has overridden this.